### DISCUSSION TITLE
Applying window style to the whole PyQt5 Application

### DISCUSSION THREAD

**Discussion by deeffest** [Akascape/py-window-styles#9] [Ideas]
I have an idea and some need to apply a window style to the whole application at once in pyqt5 for example:
```
from PyQt5.QtWidgets import QApplication, QDialog, QMainWindow, QWidget, QPushButton, QMessageBox, QHBoxLayout, QVBoxLayout

class MyWindow(QMainWindow):
    def __init__(self):
        super().__init__()

        # Create buttons and add them to the form
        button1 = QPushButton('Click me!', self)
        button1.clicked.connect(self.show_message_box)

        button2 = QPushButton('Open second window', self)
        button2.clicked.connect(self.open_second_window)

        # Arrange buttons at the center of the form
        hbox = QHBoxLayout()
        hbox.addStretch(1)
        hbox.addWidget(button1)
        hbox.addWidget(button2)
        hbox.addStretch(1)

        vbox = QVBoxLayout()
        vbox.addStretch(1)
        vbox.addLayout(hbox)
        vbox.addStretch(1)

        # Create central widget and set it for the main window
        central_widget = QWidget()
        central_widget.setLayout(vbox)
        self.setCentralWidget(central_widget)

        # Initialize the second window
        self.second_window = SecondWindow()

    def show_message_box(self):
        # Create and show the dialog box
        message_box = QMessageBox(self)
        message_box.setText('You have clicked the button!')
        message_box.exec_()

    def open_second_window(self):
        self.second_window.exec_()

class SecondWindow(QDialog):
    def __init__(self):
        super().__init__()
        self.setWindowTitle('Second Window')

if __name__ == '__main__':
    import pywinstyles
    app = QApplication([])
    pywinstyles.apply_style(app, "dark")
    window = MyWindow()
    window.show()
    app.exec_()
```
I would like QMainWindow, QMessageBox and QDialog from this code to apply the dark window style at once, since I can't apply the dark window style to the message box and in general I'm more comfortable that way. But I have no idea how hard it is to implement and if it is even possible, but I hope for the best.

**Comment by Akascape** [comment]
@deeffest Not possible at the moment. When you make any new window, you have to pass the window name to pywinstyles because every window id is unique and not interconnected.

**Reply by deeffest** [reply]
Understood, thanks for the reply! So there is no way to apply window styles to QMessageBox?

**Reply by Akascape** [reply]
I have not tested it yet. If you pass the qmessagebox window id and pass it to pywinstyles then it will work properly. You can try the last method of https://github.com/Akascape/py-window-styles/blob/main/Example_Documentation.md to get the active window.

**Reply by deeffest** [reply]
Thanks, it took me a while to figure out the solution with applying the window style to the QMessageBox:)
